vowpalwabbit/options_boost_po.h

```diff
@@ -51,6 +51,7 @@ struct options_boost_po : public options_i
   options_boost_po& operator=(options_boost_po&) = delete;
 
   void add_and_parse(const option_group_definition& group) override;
+  bool add_parse_and_check_necessary(const option_group_definition& group) override;

```

Comment:
another naming concern:

`add_parse_and_check_necessary` can do more than one thing depending on the implementation
it doesn't only add, parse, and check necessary. Could it be better represented under a more generic name, like `options.process_and_check_enabled` or `options.process_and_check_supplied`

That way when you do : `if(!options.process_and_check_enabled(new_options)) return nullptr`
 it is a bit clearer that you are returning with `nullptr` here because the reduction was not enabled. 

Also process is more generic because in the general case we are adding and parsing the options and checking whether they were all supplied, but in the name extractor case we are generating the reduction name, not really adding and parsing

Reply:
since we cannot remove (for now) add_and_parse() since it's being used to add general vw options all over the place (like in parse_args.cc). I wanted to keep the symmetry with that name to signal that it's just an augmented version of that fn call that adds a step to check if necessary was enabled. Also, I wanted the name to send a clear signal that it depends on .necessary(). Given that context, what are your thoughts?

Reply:
For me it's just not super clear when/why `options.add_parse_and_check_necessary(new_options))`  will return `false` or `true` (based on the name alone). We have the context now, but if we come back in a couple of months then we might need to go into the `add_parse_and_check_necessary` method to remember what it does and why/when it is returning `true` or `false`.

Maybe `add_parse_check_necessary_enabled`?

Reply:
That's a reasonable name, true if everything was successful (parsing, adding, enabled) false for anything else. If there's errors on parsing or adding that would be fatal errors.